run-until-success-with-backoff

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+set -euo pipefail
+
+set -x
+
+i=0
+max_retries=20
+until "$@"; do
+    ((i+=1))
+    if [[ $i == "$max_retries" ]]; then
+        echo "$@" "still failing after $max_retries retries"
+        exit 1
+    fi
+    sleep $i
```

Comment:
Do we want to move the sleep before the first attempt so we're giving some cpu time to what we're willing to wait for here?

Reply:
Done.